crates/holochain_state/src/source_chain.rs

```diff
@@ -268,13 +268,13 @@ impl SourceChain {
             .await
             .map_err(SourceChainError::other)?;
 
-            if let Err(err @ ChcError::InvalidChain(_, _)) = chc.add_records_request(payload).await
-            {
-                return Err(SourceChainError::ChcHeadMoved(
+            match chc.add_records_request(payload).await {
+                Err(e @ ChcError::InvalidChain(_, _)) => Err(SourceChainError::ChcHeadMoved(
                     "SourceChain::flush".into(),
-                    err,
-                ));
-            }
+                    e,
+                )),
+                e => e.map_err(SourceChainError::other),
+            }?;
```

Comment:
This was a bug

Reply:
Nice! I think this closes an important gap. Now we should only write to the source chain when we've also written to the CHC.

Could you add a new test to demonstrate that some other error than `InvalidChain` prevents writes?

Reply:
Did it